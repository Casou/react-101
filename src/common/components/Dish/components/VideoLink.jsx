import React from 'react';
import PropTypes from 'prop-types';
import DishLink from "./DishLink";
import IconedText from "./IconedText";
import DishType from "@/types/DishType";

const VideoLink = ({dish}) => {
  if (!dish.video) {
    return (
      <IconedText icon={"ondemand_video"}
                  title={"Youtube link"}>
        soon...
      </IconedText>
    )
  }
  return (
    <DishLink icon={"ondemand_video"}
              title={"Youtube link"}
              href={dish.video}
              text={"Video"}
    />
  )
};

VideoLink.propTypes = {
  dish: DishType.isRequired,
};
VideoLink.defaultProps = {};

export default VideoLink;