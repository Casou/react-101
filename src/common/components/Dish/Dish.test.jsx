import {render as DOMRender} from "@testing-library/react"
import snapshotRenderer from "react-test-renderer"
import VideoLink from "./components/VideoLink";

describe("Dish tests", () => {
  describe("VideoLink tests", () => {

    it("should match snapshot", () => {
      const component = snapshotRenderer.create(
        <VideoLink link="http://www.my-link.com"/>,
      )
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })

    it("should match display a link with 'Video' label if a link is passed", () => {
        const {getByText} = DOMRender(
          <VideoLink link="http://www.my-link.com"/>
        )
        const link = getByText(/Video/)
        expect(link).toHaveAttribute("href", "http://www.my-link.com")
      }
    )

    it("should match display 'soon...' with no link if no link is passed", () => {
        const {queryByText, queryByRole, getByText} = DOMRender(
          <VideoLink link=""/>
        )

        expect(queryByText(/Video/)).not.toBeInTheDocument()
        expect(() => getByText(/Video/)).toThrow() // Equivalent

        expect(queryByRole("a")).not.toBeInTheDocument()

        expect(getByText(/soon/)).toBeTruthy()
      }
    )
  })
})