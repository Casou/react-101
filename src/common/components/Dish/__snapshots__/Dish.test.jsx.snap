// Vitest Snapshot v1

exports[`Dish tests > VideoLink tests > should match snapshot 1`] = `
<div
  className="card-info"
>
  <i
    className="material-icons material-icons"
    title="Youtube link"
  >
    ondemand_video
  </i>
  <span>
    <a
      href="http://www.my-link.com"
      target="_blank"
    >
      Video
    </a>
  </span>
</div>
`;
